src/main/java/io/temporal/samples/hello/HelloDetachedCancellationScope.java

```diff
@@ -0,0 +1,226 @@
+            ActivityOptions.newBuilder()
+                .setStartToCloseTimeout(Duration.ofSeconds(10))
+                .setCancellationType(ActivityCancellationType.WAIT_CANCELLATION_COMPLETED)
+                .build());
+
+    @Override
+    public String getGreeting(String name) {
+      try {
+        this.greeting = activities.sayHello(name);
+        return greeting;
+      } catch (ActivityFailure af) {
+        // Create a CancellationScope that is not linked to a parent scope
+        // This can be used to do cleanup code after workflow has been cancelled
+        CancellationScope detached =
+            Workflow.newDetachedCancellationScope(() -> greeting = activities.sayGoodBye(name));
+        detached.run();
+        throw af;
+      }
+    }
+
+    @Override
+    public String queryGreeting() {
+      return greeting;
+    }
+  }
+
+  public static void main(String[] args) throws InterruptedException {
+
+    // Define the workflow service.
+    WorkflowServiceStubs service = WorkflowServiceStubs.newInstance();
+
+    /**
+     * Define the workflow client. It is a Temporal service client used to start, signal, and query
+     * workflows
+     */
+    WorkflowClient client = WorkflowClient.newInstance(service);
+
+    /**
+     * Define the workflow factory. It is used to create workflow workers for a specific task queue.
+     */
+    WorkerFactory factory = WorkerFactory.newInstance(client);
+
+    /**
+     * Define the workflow worker. Workflow workers listen to a defined task queue and process
+     * workflows and activities.
+     */
+    Worker worker = factory.newWorker(TASK_QUEUE);
+
+    /**
+     * Register our workflow implementation with the worker. Workflow implementations must be known
+     * to the worker at runtime in order to dispatch workflow tasks.
+     */
+    worker.registerWorkflowImplementationTypes(GreetingWorkflowImpl.class);
+
+    /**
+     * Register our workflow activity implementation with the worker. Since workflow activities are
+     * stateless and thread-safe, we need to register a shared instance.
+     */
+    worker.registerActivitiesImplementations(new GreetingActivitiesImpl());
+
+    /**
+     * Start all the workers registered for a specific task queue. The started workers then start
+     * polling for workflows and activities.
+     */
+    factory.start();
+
+    // Create the workflow client stub. It is used to start our workflow execution.
+    GreetingWorkflow workflow =
+        client.newWorkflowStub(
+            GreetingWorkflow.class,
+            WorkflowOptions.newBuilder()
+                .setWorkflowId(WORKFLOW_ID)
+                .setTaskQueue(TASK_QUEUE)
+                .build());
+
+    WorkflowClient.start(workflow::getGreeting, "John");
+
+    WorkflowStub workflowStub = WorkflowStub.fromTyped(workflow);
+
+    Thread.sleep(1000);
+
+    // Cancel Workflow execution
+    // This can be done from a different client for example
```

Comment:
```suggestion
    // Cancel the Workflow Execution
    // Note that this can be done from a different client.
```

Because everything is in this one file, I am not sure this is super helpful.
In most cases a Workflow Execution would likely be canceled from a process OTHER than the one that started it.
So, basically all this code is repetitive up until this call which shows `workflowStub.cancel()`.